packages/contracts/contracts/standards/L2StandardERC721.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import { ERC721 } from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import { IERC165 } from "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+import { Strings } from "@openzeppelin/contracts/utils/Strings.sol";
+import { Lib_PredeployAddresses } from "../libraries/constants/Lib_PredeployAddresses.sol";
+import { Lib_Strings } from "../libraries/utils/Lib_Strings.sol";
+import "./IL2StandardERC721.sol";
+
+contract L2StandardERC721 is IL2StandardERC721, ERC721 {
+    address public l1Token;
+    address public l2Bridge;
+    string public baseTokenURI;
+
+    /**
+     * @param _l2Bridge Address of the L2 standard bridge.
+     * @param _l1Token Address of the corresponding L1 token.
+     * @param _name ERC721 name.
+     * @param _symbol ERC721 symbol.
+     */
+    constructor(
+        address _l2Bridge,
+        address _l1Token,
+        string memory _name,
+        string memory _symbol
+    ) ERC721(_name, _symbol) {
+        l1Token = _l1Token;
+        l2Bridge = _l2Bridge;
+
+        // Creates a base URI in the format specified by EIP-681:
+        // https://eips.ethereum.org/EIPS/eip-681
+        baseTokenURI = string(
+            abi.encodePacked(
+                "ethereum:0x",
+                Lib_Strings.addressToString(_l1Token),
+                "@",
+                Strings.toString(block.chainid),
+                "/tokenURI?uint256="
+            )
+        );
+    }
+
+    modifier onlyL2Bridge() {
+        require(msg.sender == l2Bridge, "Only L2 Bridge can mint and burn");
+        _;
+    }
+
+    // slither-disable-next-line external-function
+    function supportsInterface(bytes4 _interfaceId)
+        public
+        view
+        override(ERC721, IERC165)
+        returns (bool)
+    {
+        bytes4 firstSupportedInterface = bytes4(keccak256("supportsInterface(bytes4)")); // ERC165
```

Comment:
I think our recent audit said this was a complex way to compute the interface, cc @maurelian

Reply:
Honestly I think it's fine, but this style is a bit cleaner, and for some reason slightly cheaper: 

https://github.com/ethereum-optimism/optimistic-specs/commit/da14040fe518e0a0662c5718c868313db41ddc11#diff-1d13916238a897be0796ef4083fd1fbde156026d4f2631d9743996eda9339f28R76-R79

Reply:
Updated this to the cleaner style